lib/cpp/src/thrift/concurrency/Mutex.h

```diff
@@ -101,25 +101,6 @@ class ReadWriteMutex {
   boost::shared_ptr<impl> impl_;
 };
 
-/**
- * A ReadWriteMutex that guarantees writers will not be starved by readers:
- * When a writer attempts to acquire the mutex, all new readers will be
- * blocked from acquiring the mutex until the writer has acquired and
- * released it. In some operating systems, this may already be guaranteed
- * by a regular ReadWriteMutex.
- */
-class NoStarveReadWriteMutex : public ReadWriteMutex {
-public:
-  NoStarveReadWriteMutex();
-
-  virtual void acquireRead() const;
-  virtual void acquireWrite() const;
-
-private:
-  Mutex mutex_;
-  mutable volatile bool writerWaiting_;
-};
```

Comment:
Item #1 in the original pull request indicates that NoStarveReadWriteMutex may not be correct; given it is not used by the project and is not a core Thrift mechanism I see no need to keep it because it increases the maintenance burden of the project.  I will use a separate pull request to remove them instead.